persistent/tests/test_picklecache.py

```diff
@@ -1096,33 +1075,45 @@ def _p_deactivate(self):
 class CPickleCacheTests(PickleCacheTests):
 
     def _getTargetClass(self):
-        from persistent.cPickleCache import PickleCache
-        return PickleCache
+        from persistent._compat import _c_optimizations_available as get_c
+        return get_c()['persistent.picklecache'].PickleCache
 
     def _getDummyPersistentClass(self):
         from persistent import Persistent
         class DummyPersistent(Persistent):
             __slots__ = ()
         return DummyPersistent
 
+    def test_inst_does_not_conform_to_IExtendedPickleCache(self):
+        from persistent.interfaces import IExtendedPickleCache
+        from zope.interface.verify import verifyObject
+        from zope.interface.exceptions import DoesNotImplement
+        from zope.interface.exceptions import BrokenImplementation
+        # We don't claim to implement it.
+        with self.assertRaises(DoesNotImplement):
+            verifyObject(IExtendedPickleCache, self._makeOne())
+        # And we don't even provide everything it asks for.
+        with self.assertRaises(BrokenImplementation):
+            verifyObject(IExtendedPickleCache, self._makeOne(), tentative=True)
```

Comment:
That seems like a strange thing to test for.

Reply:
Ah, I think I get it: it's to make sure the `@use_c_impl` doesn't accidentally copy the IExtendedPickleCache declaration onto the C version of the pickle cache that doesn't actually implement it!

Reply:
Exactly. Seems like it could use an explanatory comment.